modules/packages/UnrolledLinkedList.chpl

```diff
@@ -0,0 +1,1249 @@
+    }
+
+    /*
+      Remove the element at the index `idx` from this unrolledLinkedList and return it.
+
+      .. warning::
+
+        Popping an element from this unrolledLinkedList will invalidate any reference to
+        the element taken while it was contained in this unrolledLinkedList.
+
+      .. warning::
+
+        Calling this method on an empty unrolledLinkedList or with values of `idx` that
+        are out of bounds will cause the currently running program to halt.
+        If the `--fast` flag is used, no safety checks will be performed.
+
+      :arg idx: The index of the element to remove.
+      :type idx: `int`
+
+      :return: The element popped.
+      :rtype: `eltType`
+    */
+    proc ref pop(idx: int): eltType {
+      _enter();
+      var result = _popAtIndex(idx);
+      _leave();
+      return result;
+    }
+
+    /*
+      Clear the contents of this unrolledLinkedList.
+
+      .. warning::
+
+        Clearing the contents of this unrolledLinkedList will invalidate all existing
+        references to the elements contained in this unrolledLinkedList.
+    */
+    proc ref clear() {
+      on this {
+        _enter();
+
+        _destroy();
+        _head = nil;
+        _tail = nil;
+        _size = 0;
+
+        _sanity(_size == 0);
+        _sanity(_head == nil);
+
+        _leave();
+      }
+    }
+
+    /*
+      Return a zero-based index into this unrolledLinkedList of the first item whose value
+      is equal to `x`. If no such element can be found this method returns
+      the value `-1`.
+
+      .. warning::
+
+        Calling this method on an empty unrolledLinkedList or with values of `start` or 
+        `end` that are out of bounds will cause the currently running program
+        to halt. If the `--fast` flag is used, no safety checks will be
+        performed.
+
+      :arg x: An element to search for.
+      :type x: `eltType`
+
+      :arg start: The start index to start searching from.
+      :type start: `int`
+
+      :arg end: The end index to stop searching at. A value less than
+                `0` will search the entire unrolledLinkedList.
+      :type end: `int`
+
+      :return: The index of the element to search for, or `-1` on error.
+      :rtype: `int`
+    */
+    proc const indexOf(x: eltType, start: int=0, end: int=-1): int {
+      if boundsChecking {
+        const msg = " index for \"unrolledLinkedList.indexOf\" out of bounds: ";
+
+        if end >= 0 && !_withinBounds(end) then
+          boundsCheckHalt("End" + msg + end:string);
+
+        if !_withinBounds(start) then
+          boundsCheckHalt("Start" + msg + start:string);
+      }
+
+      param error = -1;
+
+      if end >= 0 && end < start then
+        return error;
+
+      var result = error;
+
+      on this {
+        _enter();
+
+        const stop = if end < 0 then _size-1 else end;
+
+        var i = 0;
+        for item in this {
+          if i > stop then break;
+          if i >= start {
```

Comment:
Do you suggest adding a new interface? Or just a refactor? Also, I didn't see the benefit here. The iterating here is straightforward and I suppose adding a function here will add up to the complexity.

Reply:
I think something like that can be added later on. It can help jump over ULL nodes until the start is hit, instead of iterating element-by-element until `start`

Reply:
That's what I was thinking, but even if there were agreement on it, it's beyond the scope of this PR.

Reply:
Huh.  I'd mark this thread resolved, but I can't find the UI for doing that.  I assume it's supposed to be where the `Show resolved` is on resolved threads?